agent/agent/main.go

```diff
@@ -38,6 +38,8 @@ func main() {
 		fmt.Println("Must specify rootUrl.")
 		return
 	}
+
+	cred := common.MakeCredentialFromArgs(username, password)
```

Comment:
I changed it to MakeCredentialFromCliArgs in a commit that wasn't pushed. Sigh.

The idea here was that CLI runner will collect whatever information it collects from cli flags, environment, etc. and pass it all into this method. It will then take care of assembling the right Credential structure. This is just a few lines of copy/paste saved between services, that's all.